src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -22,6 +24,26 @@
 from pants.rules.core.strip_source_root import SourceRootStrippedSources
 
 
+def calculate_timeout_seconds(
+  *,
+  timeouts: bool,
+  test_target_timeout_seconds: Optional[int],
+  timeout_default_seconds: Optional[int],
+  timeout_maximum_seconds: Optional[int],
```

Comment:
You signal in the function signature that this deals with seconds, so it's now repetitive to specify it in each param name too. I recommend dropping `_seconds` from the param names to improve readability.